### PR TITLE
feat: adding vaul vue integration, closes #370

### LINKED ISSUE #370: [Feature]: Drawer (vaul-vue)
[Vaul-vue](https://github.com/radix-vue/vaul-vue/releases) should be ready to go now. 🤩

Appreciate the hardwork from @Elliot-Alexander ! 💚

### PR DESCRIPTION
**Pull request by Saeid-Za**
There are still 2 remaining tasks here:
1- Add responsive combo-box example
2- make ThemeCustomizer responsive via `Drawer`

Closes #370

### CODE REVIEW

**Comment by sadeghbarati** [comment]
Thanks, Saeid 🙌 

Will review it shortly

---

If you still have to work on it feel free to mark it as `draft` and do it in your free time 💯 
Search for `Still in progress?` or `Convert to draft`

File: apps/www/src/lib/registry/default/example/DrawerDialog.vue [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      </div>
+      <div class="grid gap-2">
+        <Label html-for="username">Username</Label>
+        <Input id="username" default-value="@shadcn" />
+      </div>
+      <Button type="submit">
+        Save changes
+      </Button>
+    </form>
+  </UseTemplate>
```
Comment (sadeghbarati) [inline apps/www/src/lib/registry/default/example/DrawerDialog.vue]: Can you move `UseTemplate` to the start of `<template>` just like VueUse docs?

IIRC they said it's better to add `DefineTemplate` (UseTemplate) **first** and then use `ReuseTemplate` (GridForm)

After these changes please run `build:registry` again thanks!
Also for the next time please create a new branch it's better not to add changes in the dev branch 🙏

**Comment by Saeid-Za** [comment]
> Thanks, Saeid 🙌
> 
> Will review it shortly
> 
> If you still have to work on it feel free to mark it as `draft` and do it in your free time 💯 Search for `Still in progress?` or `Convert to draft`

Happy to help!
The rest of the changes that are remaining are not blocking this PR, I'll add an other PR addressing those later today. but if they are required, I'll change the status to `draft`.

File: apps/www/src/lib/registry/default/example/DrawerDialog.vue (hunk 1 above)
Comment (Saeid-Za) [reply apps/www/src/lib/registry/default/example/DrawerDialog.vue]: Done, there were no additional changes after I had built the registry.

**Comment by sadeghbarati** [comment]
@Saeid-Za Nice, ok then run `build:registry` in `apps/www` and let's merge this PR 🎉 

And do these in another PR 🙏 

>There are still 2 remaining tasks here:
1- Add responsive combo-box example
2- make ThemeCustomizer responsive via Drawer

**Comment by Saeid-Za** [comment]
> @Saeid-Za Nice, ok then run `build:registry` in `apps/www` and let's merge this PR 🎉 
> 
> And do these in another PR 🙏 
> 
> >There are still 2 remaining tasks here:
> 1- Add responsive combo-box example
> 2- make ThemeCustomizer responsive via Drawer

It seems that I'd already built the registry and is avalable in previous commit and forgot to mention it 😅

**Comment by sadeghbarati** [comment]
Thanks!

**Pull request merged**